### PR TITLE
add unit tests for addons/functions/

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Add unit tests for addons/functions/

# Impacts
Venom unit tests

# Delete branch after merge
YES

### CODE REVIEW

**Comment by julsemaan** [comment]
See this pipeline: https://gitlab.com/inverse-inc/packetfence/-/pipelines/456189959

File: t/addons-functions/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Edit this file and fill the different variables
+
+## Running the tests
+
+Once you're setup, just run:
+
+```
+cd addons/functions/t/
+./run-all.tests
```
Comment (nqb) [inline t/addons-functions/README.md]:
```suggestion
make test
```

File: t/venom/test_suites/unit_tests/addons_functions_tests.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  - type: exec
+    script: "sed -i 's#%%TEST_INTERFACE%%#{{ .pfserver_mgmt_interface }}#g' /usr/local/pf/t/addons-functions/vars.tests"
+  - type: exec
+    script: "sed -i 's#%%TEST_INTERFACE_IP%%#{{ .pfserver_mgmt_ip }}#g' /usr/local/pf/t/addons-functions/vars.tests"
+  - type: exec
+    script: "sed -i 's#%%TEST_INTERFACE_CIDR%%#{{ .pfserver_mgmt_cidr }}#g' /usr/local/pf/t/addons-functions/vars.tests"
+  - type: exec
+    script: "sed -i 's#%%TEST_INTERFACE_NETMASK%%#{{ .pfserver_mgmt_netmask }}#g' /usr/local/pf/t/addons-functions/vars.tests"
+  - type: exec
+    script: "sed -i 's#%%TEST_INTERFACE_NETWORK%%#{{ .pfserver_mgmt_net }}#g' /usr/local/pf/t/addons-functions/vars.tests"
```
Comment (nqb) [inline t/venom/test_suites/unit_tests/addons_functions_tests.yml]: Could you remove space inside curly brackets ?

Comment (nqb) [reply t/venom/test_suites/unit_tests/addons_functions_tests.yml]: If you rebase on devel, you can use `pfserver_test_dir` variable to replace `/usr/local/pf/t`

**Review by nqb — CHANGES_REQUESTED** [review]
- [X] You need to adjust packaging for `packetfence-test` on Debian like you did for EL
- [X] I think `vars.tests.ci-el8` could be rename `vars.tests.ci` because I don't think why it would be different at some point for Debian

I will review *.tests later

**Review by nqb — COMMENTED** [review]
If I understand correctly the code;

`failed_test "get_os_ip_address provides the right IP address"` means that ` test_get_os_ip_address` **doesn't** provide the right IP address no ?

**Comment by julsemaan** [comment]
> If I understand correctly the code;
> 
> `failed_test "get_os_ip_address provides the right IP address"` means that ` test_get_os_ip_address` **doesn't** provide the right IP address no ?

Correct so it fails the assertion `get_os_ip_address provides the right IP address`

The text describes what **should** happen and if it doesn't the output will be:
`FAILED: get_os_ip_address provides the right IP address`

**Comment by julsemaan** [comment]
> * You need to adjust packaging for `packetfence-test` on Debian like you did for EL

I'm not sure what/where to adjust that since unit tests only seem to be running on EL8 from what I saw

> * I think `vars.tests.ci-el8` could be rename `vars.tests.ci` because I don't think why it would be different at some point for Debian
> 

Agreed, I'll adjust that

**Comment by nqb** [comment]
> I'm not sure what/where to adjust that since unit tests only seem to be running on EL8 from what I saw

You're right regarding unit tests running on EL8 but we packaged `packetfence-test` for Debian too. I will adjust packaging for Debian.

**Pull request merged**